TYPO3.Neos/Classes/TYPO3/Neos/EventLog/Domain/Service/EventEmittingService.php

```diff
@@ -171,13 +179,20 @@ protected function getCurrentContext()
     }
 
     /**
-     * Set the current account identifier
+     * Try to set the current account identifier emitting the events, if possible
      *
-     * @param string $accountIdentifier
      * @return void
      */
-    public function setCurrentAccountIdentifier($accountIdentifier)
+    protected function initializeAccountIdentifier()
     {
-        $this->currentAccountIdentifier = $accountIdentifier;
+        if (isset($this->currentAccountIdentifier)) {
+            return;
+        }
+        if ($this->securityContext->canBeInitialized()) {
+            $account = $this->securityContext->getAccount();
```

Comment:
You just moved this here, but it should be noted that `Security\Context::getAccount()` only returns the first authenticated account whichever this is. `getAccountByAuthenticationProviderName()` would prevent a "3rd party" account from being returned. Alternatively it might be possible to use the `UserService`, which already has this filter built in

Reply:
good point, should I fix that while at it?